ios/RNSScreenStackHeaderSubview.mm

```diff
@@ -22,6 +22,15 @@ @implementation RNSScreenStackHeaderSubview
 
 #pragma mark - Common
 
+// We're forcing the navigation controller's view to re-layout
+// see: https://github.com/software-mansion/react-native-screens/pull/2385
+- (void)ensureNavigationLayout
+{
+  UIViewController *vc = self.reactSuperview.reactSuperview.reactViewController;
+  UINavigationController *navctr = vc.navigationController;
```

Comment:
Let's rewrite this, so the code is more readable.

`self.reactSuperview` should be a `RNSScreenStackHeaderConfig` right? Create a method that gets it. `HeaderConfig` does have `screenView` field, from which you can access `reactViewController` and then the navigation controller.

```suggestion
  UINavigationController *navctr = self.headerConfig.screenView.reactViewController.navigationController;
```

It would be also ok, to extract it to separate method.

Reply:
Referencing headerConfig makes sense! Added a getter method for the header, but I don't think we'll be reusing those methods tbh. https://github.com/software-mansion/react-native-screens/pull/2385/commits/282b86d642a49dad764db493d5f651a9d0b6ddb7